src/msix/pack/AppxPackageWriter.cpp

```diff
@@ -2,26 +2,60 @@
 //  Copyright (C) 2019 Microsoft.  All rights reserved.
 //  See LICENSE file in the project root for full license information.
 // 
+
+#include "AppxPackaging.hpp"
 #include "AppxPackageWriter.hpp"
 #include "MsixErrors.hpp"
 #include "Exceptions.hpp"
+#include "ContentType.hpp"
+#include "SHA256.hpp"
 
 #include <string>
+#include <memory>
+#include <future>
+#include <algorithm>
+#include <functional>
 
 namespace MSIX {
 
+    static const std::uint32_t defaultBlockSize = 65536;
+
     AppxPackageWriter::AppxPackageWriter(IStream* outputStream) : m_outputStream(outputStream)
     {
         m_state = WriterState::Open;
+        m_blockMapWriter = std::make_unique<BlockMapWriter>();
+        m_contentTypeWriter = std::make_unique<ContentTypeWriter>();
     }
 
     // IPackageWriter
     void AppxPackageWriter::Pack(const ComPtr<IDirectoryObject>& from)
     {
         ThrowErrorIf(Error::InvalidState, m_state != WriterState::Open, "Invalid package writer state");
         auto fileMap = from->GetFilesByLastModDate();
+        std::vector<std::unique_ptr<PayloadFile>> files;
+        for(const auto& file : fileMap)
+        {
+            std::string ext = file.second.substr(file.second.find_last_of(".") + 1);
+            std::transform(ext.begin(), ext.end(), ext.begin(), ::tolower);
+            std::string contentType;
+            APPX_COMPRESSION_OPTION compressionOpt;
+            auto findExt = extToContentType.find(ext);
+            if (findExt == extToContentType.end())
+            {
+                // if it's not in the map these are the defaults

```

Comment:
The interface should be something more like:

const ContentTypeData& GetExtensionContentTypeData(string ext)

And it would handle the map lookup and the defaulting.